### PR TITLE
Removed getattr() from some settings accesses.

### PR DESCRIPTION
**Pull request by adamchainz**
# Trac ticket number
N/A

# Branch description

The modified lines all read settings defined [in `global_settings.py`](https://github.com/django/django/blob/main/django/conf/global_settings.py), which is always copied from when creating the settings object: https://github.com/django/django/blob/549320946dfdaac6ccb6c0d3150c46db92637187/django/conf/__init__.py#L159-L161

# Checklist
- [x] This PR targets the `main` branch. 
- [x] The commit message is written in past tense, mentions the ticket number, and ends with a period.
- [n/a] I have checked the "Has patch" **ticket flag** in the Trac system.
- [x] I have added or updated relevant **tests**.
- [x] I have added or updated relevant **docs**, including release notes if applicable.
- [x] For UI changes, I have attached **screenshots** in both light and dark modes.

### CODE REVIEW

File: tests/middleware/tests.py [hunk 1]
```diff
@@ -750,10 +749,8 @@ def test_defaults_sameorigin(self):
         If the X_FRAME_OPTIONS setting is not set then it defaults to
         DENY.
         """
-        with override_settings(X_FRAME_OPTIONS=None):
-            del settings.X_FRAME_OPTIONS  # restored by override_settings
```
Comment (adamchainz) [inline tests/middleware/tests.py:754]: This line was testing an artificial situation where `del` was used to remove the setting entirely. If we want to support that, I think we need to support it for *all* settings access, which would be too much.

**Comment by kezabelle** [comment]
> The modified lines all read settings defined in `global_settings.py`

🤔 I don't think that's _actually_ guaranteed, because you can do:
```python
from django.conf import settings
settings.configure(..., default_settings={})
```
to effectively not bring in any Django defaults.

I'm not sure if that's a _problem_ in this scenario (the tests pass ...), or if it's just so niche it doesn't matter 🤷

**Comment by adamchainz** [comment]
Huh, TIL. But then don’t *all* settings accesses need `getattr()` ? That seems rather unuseful and not what is done in most of Django.

**Comment by felixxm** [comment]
TBH, I don't think it's worth doing. I know that we're not consistent with using `getattr()` on `settings` but any change to this behavior now would be backward incompatible.

**Comment by adamchainz** [comment]
Yeah, that’s fair. Going forwards, though, I guess we’ll continue to use `settings.<NAME>`? It seems fair to me that if you are using a different `default_setitngs`, you provide all the settings your project needs.

**Pull request closed without merging**